### PR TITLE
Stories/etl for relationship to hoh

### PR DESCRIPTION
**Pull request by eanders**

### CODE REVIEW

File: drivers/hmis_csv_twenty_twenty/app/models/hmis_csv_twenty_twenty/hmis_csv_cleanup/enforce_relationship_to_hoh.rb [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+  class EnforceRelationshipToHoh < Base
+    def cleanup!
+      # Set anyone with a blank HouseholdID to RelationshipToHoH = 1
+      # These clients are not in households
+      enrollment_scope.where(HouseholdID: nil).
+        where(
+          ie_t[:RelationshipToHoH].not_eq(1).
+          or(ie_t[:RelationshipToHoH].eq(nil)),
+        ).
+        update_all(RelationshipToHoH: 1)
```
Comment (awisema) [inline drivers/hmis_csv_twenty_twenty/app/models/hmis_csv_twenty_twenty/hmis_csv_cleanup/enforce_relationship_to_hoh.rb]: Do you need to give these clients HouseholdIDs?

File: drivers/hmis_csv_twenty_twenty/app/models/hmis_csv_twenty_twenty/hmis_csv_cleanup/enforce_relationship_to_hoh.rb [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+      end
+
+      enrollment_scope.where(HouseholdID: individual_household_ids).
+        where(
+          ie_t[:RelationshipToHoH].not_eq(1).
+          or(ie_t[:RelationshipToHoH].eq(nil)),
+        ).
+        update_all(RelationshipToHoH: 1)
+
+      enrollment_scope.where(HouseholdID: multi_person_to_fix.keys).
```
Comment (awisema) [inline drivers/hmis_csv_twenty_twenty/app/models/hmis_csv_twenty_twenty/hmis_csv_cleanup/enforce_relationship_to_hoh.rb:95]: `# Make any HoH's 99's, leaving any other relationships -- will add HoH below`

File: drivers/hmis_csv_twenty_twenty/app/models/hmis_csv_twenty_twenty/hmis_csv_cleanup/enforce_relationship_to_hoh.rb [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+            multi_person_to_fix[hh_id] = oldest_adult[:row_id]
+          elsif child_under_11
+            multi_person_to_fix[hh_id] = oldest_client[:row_id]
+          else
+            multi_person_with_no_child_under_11 += rows
+          end
+        end
+      end
+
+      enrollment_scope.where(HouseholdID: individual_household_ids).
```
Comment (awisema) [inline drivers/hmis_csv_twenty_twenty/app/models/hmis_csv_twenty_twenty/hmis_csv_cleanup/enforce_relationship_to_hoh.rb:72]: `# Make sure all individuals are HoH`

File: drivers/hmis_csv_twenty_twenty/app/models/hmis_csv_twenty_twenty/hmis_csv_cleanup/enforce_relationship_to_hoh.rb [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+# If the household only has one person, ensure RelationshipToHoH is 1
+# If the household has more than one person, and there is a female 18 or older,
+# set the oldest female to RelationshipToHoH = 1 and set any other RelationshipToHoH == 1 to 99
+# If the household has more than one person, but no female 18 or older, and someone in the household is 18 or older, set the oldest person to RelationshipToHoH = 1 and set any other RelationshipToHoH == 1 to 99
+# If everyone in the household is 17 or younger, and there is a person 10 or younger, set the oldest person to RelationshipToHoH = 1 and set any other RelationshipToHoH == 1 to 99
+# If the household only contains clients between the age of 11 and 17 inclusive, break up the household and set everyone as RelationshipToHoH = 1
+
+module HmisCsvTwentyTwenty::HmisCsvCleanup
+  class EnforceRelationshipToHoh < Base
+    def cleanup!
```
Comment (awisema) [inline drivers/hmis_csv_twenty_twenty/app/models/hmis_csv_twenty_twenty/hmis_csv_cleanup/enforce_relationship_to_hoh.rb:18]: Any touched enrollments needs the source hash updated.

**Review by awisema — CHANGES_REQUESTED** [review]

**Comment by eanders** [comment]
@awisema, I _think_ I got this right this time.

**Review by awisema — APPROVED** [review]
Looks good to me! And there are even tests!

**Pull request merged**